builtin/mainmenu/tab_online.lua

```diff
@@ -210,21 +210,66 @@ local function search_server_list(input)
 	for i = 1, #serverlistmgr.servers do
 		local server = serverlistmgr.servers[i]
 		local found = 0
+		local sername_keywords = {}
+		local sername_keywords_matchtimes = {}
+		local server_desc_keywords = {}
+		local server_desc_keywords_matchtimes = {}
+		-- Make search precise for exactly matched server
+		-- Or at least more exactly matched server should be rated higher at the match scoring
+		if server.name then
+			for word in server.name:gmatch("%S+") do
+				word = word:gsub("(%W)", "%%%1"):lower()
```

Comment:
```suggestion
				word = word:gsub("%W", "%%%1"):lower()
```

Also at this point probably call it `local pattern` because this is an escaped pattern.

Reply:
After reading the rest of the code, I don't see the point of this: Why are you escaping something which is, as far as I can tell, never used as a pattern? I assume you're doing so because the keywords are escaped, but then the proper solution is just to remove the code that escapes the keywords.